src/shared/lib/observables/interimSubmitObservable.ts

```diff
@@ -0,0 +1,31 @@
+import { CommonObservable } from '../utils';
+
+export interface IInterimSubmitObservable {
+  notify: () => void;
+}
+
+class InterimSubmitObservable
+  extends CommonObservable
+  implements IInterimSubmitObservable
+{
+  constructor() {
+    super();
+  }
+
+  private _processing: boolean = false;
+
+  public get processing() {
```

Comment:
Please implement `IMutex` interface instead of `processing` getter and `reset` because it does exactly the same thing

Reply:
Mutex doesn't help here. Where would you suggest to place it? 
We change observable state from the Interim screen. Should Mutex be there? I don't think so, as mutex usually bound with any instance of any class.

Reply:
1st looks good.